pro/src/ui-kit/form/TextInputAutocomplete/TextInputAutocomplete.tsx

```diff
@@ -58,24 +62,42 @@ const AutocompleteTextInput = ({
   const [lastSelectedValue, setLastSelectedValue] =
     useState<IAutocompleteItemProps>()
 
-  const { setFieldValue } = useFormikContext()
+  const { setFieldValue, setFieldTouched } = useFormikContext()
 
-  const updateSuggestions = (search: string) => {
-    const suggestions = getSuggestions(search)
-    if (suggestions.length > 0) {
-      setSuggestions(suggestions)
-      setIsOpen(true)
-    } else {
-      setIsOpen(false)
-    }
-  }
-  const handleSearchChange = (search: string) => {
-    updateSuggestions(search)
-    if (!search) {
-      setIsOpen(false)
-    }
-    setFieldValue(`search-${fieldName}`, search, false)
-  }
+  const updateSuggestions = useCallback(
+    async (search: string) => {
+      const suggestions = await getSuggestions(search)
+      if (suggestions.length > 0) {
+        setSuggestions(suggestions)
+        if (isFocus) {
+          setIsOpen(true)
+        }
+      } else {
+        setIsOpen(false)
+      }
+    },
+    [isFocus]
+  )
+
+  const debouncedSearch = useCallback(debounce(updateSuggestions, 300), [
+    isFocus,
+  ])
+
+  const handleSearchChange = useCallback(
+    (search: string) => {
+      if (useDebounce) {
+        debouncedSearch(search)
+      } else {
+        updateSuggestions(search)
+      }
+
+      if (!search) {
+        setIsOpen(false)
+      }
+      setFieldValue(`search-${fieldName}`, search, false)
+    },
+    [updateSuggestions]
```

Comment:
```suggestion
    [updateSuggestions]
```
pas besoin de `debouncedSearch` ici ?
(j'avoue que je sais jamais trop ce qu'il faut mettre)